### PR TITLE
docs: kms-apis: add documentation for the KMS APIs exposed by the SDK

### PR DESCRIPTION
**Pull request by petreeftime**
Signed-off-by: Petre Eftime <epetre@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/kms-apis/kms-apis.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+These APIs allow an AWS Nitro Enclave to decrypt, encrypt (via `GenerateDataKey`) data
+and generate a random number if they match the AWS KMS policy set in place, and provide a 
+response from AWS KMS that only the enclave can use.
+
+Cryptographic attestation allows restricting access to an AWS KMS key based on a set of
+additional conditions that can be verified only by providing a valid `AttestationDocument`.
+Access is then given if the `AttestationDocument` is validly signed by the Nitro hypervisor, 
+it contains a public key and the PCRs listed in the AttestationDocument match the PCRs
+in the policy. An [example policy](#example-policy) is provided below, which allows the role `ROLE_ARN` to
+use the decrypt operation if it can provide a valid AttestationDocument containing
```
Comment (ddraga) [inline docs/kms-apis/kms-apis.md:34]: For the sake of being consistent I would put every occurrence of "AttestationDocument" between back quotes

File: docs/kms-apis/Decrypt.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+* [GenerateDataKeyPairWithoutPlaintext](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKeyPairWithoutPlaintext.html)
+
+You can use this operation to decrypt ciphertext that was encrypted under a symmetric or asymmetric CMK. When the CMK is asymmetric, you must specify the CMK and the encryption algorithm that was used to encrypt the ciphertext. For information about symmetric and asymmetric CMKs, see [Using Symmetric and Asymmetric CMKs](https://docs.aws.amazon.com/kms/latest/developerguide/symmetric-asymmetric.html) in the *AWS Key Management Service Developer Guide*.
+
+The Decrypt operation also decrypts ciphertext that was encrypted outside of AWS KMS by the public key in an AWS KMS asymmetric CMK. However, it cannot decrypt ciphertext produced by other libraries, such as the [AWS Encryption SDK](https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/) or [Amazon S3 client-side encryption](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html). These libraries return a ciphertext format that is incompatible with AWS KMS.
```
Comment (ddraga) [inline docs/kms-apis/Decrypt.md:14]: "The Decrypt operation also" -> I would put "Decrypt" between backquotes

File: docs/kms-apis/Decrypt.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+Type: String
+
+Length Constraints: Minimum length of 1. Maximum length of 2048.
+
+Required: No
+
+#### Recipient
+
+`Recipient` is mentioned in the request then the response will contain `CiphertextForRecipient` instead of `Plaintext`, and the response will be encrypted with the public key included in `AttestationMetadata`.
```
Comment (ddraga) [inline docs/kms-apis/Decrypt.md:127]: I would start this phrase with "When".

File: docs/kms-apis/GenerateDataKey.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+Type: Integer
+
+Valid Range: Minimum value of 1. Maximum value of 1024.
+
+Required: No
+
+#### Recipient
+
+`Recipient` is mentioned in the request then the response will contain `CiphertextForRecipient` instead of `Plaintext`, and the response will be encrypted with the public key included in `AttestationMetadata`.
```
Comment (ddraga) [inline docs/kms-apis/GenerateDataKey.md:144]: Again, I would start the phrase with "When".

File: docs/kms-apis/RecipientRequest.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# RecipientRequest
+
+Specifies that the data is intented for a particular Recipient and must be re-encrypted using the specified private key before being sent back.
```
Comment (andraprs) [inline docs/kms-apis/RecipientRequest.md:3]: s/private key/public key/

**Review by andraprs — APPROVED** [review]

**Review by ddraga — APPROVED** [review]

**Review by alcioa — APPROVED** [review]

**Pull request merged**